### PR TITLE
Support JA locale

### PR DESCRIPTION
**Pull request by technohippy**
I'm adding resources for JA locale.

### CODE REVIEW

**Comment by kepae** [comment]
Hey @technohippy -- thanks for the PR, this is great! Would love to help Japanese readers.

Feel free to let us know when this is ready to be reviewed so that we can deploy a preview of the changes on Netlify.
(I note that some changed files are not entirely translated, [e.g.](https://github.com/responsible-ai-collaborative/aiid/blob/e60122a73be53a88b072c3efbb0328b68a991e02/site/gatsby-site/i18n/locales/ja/submit.json))

**Comment by technohippy** [comment]
@kepae Translation's been done. Could you review this PR?

**Comment by kepae** [comment]
I reviewed the content (with a ~little~ lot of help from DeepL and Google Translate).

However, the deploy I enabled failed due to tests not passing, and then exceeding the Netlify build limit due to time needed to retry tests:
https://app.netlify.com/sites/staging-aiid/deploys/65701f3345af4700084f5fee

```
10:43:28 AM:   1) Edit report
10:43:28 AM:        Should load and update report values:
10:43:28 AM:      TypeError: The following error originated from your application code, not from Cypress.
10:43:28 AM:   > Cannot read properties of undefined (reading "title")
10:43:28 AM: When Cypress detects uncaught errors originating from your application it will automatically fail the current test.
10:43:28 AM: This behavior is configurable, and you can choose to turn this off by listening to the `uncaught:exception` event.
10:43:28 AM: https://on.cypress.io/uncaught-exception-from-application
10:43:28 AM:       at <unknown> (http://localhost:8080/component---src-pages-cite-edit-js-445fcac5ccff4fc2c127.js:1:41888)
10:43:28 AM:       at Array.map (<anonymous>)
10:43:28 AM:       at C (http://localhost:8080/component---src-pages-cite-edit-js-445fcac5ccff4fc2c127.js:1:41457)
10:43:28 AM:       at Eo (http://localhost:8080/framework-6858c50a67608e0b80d0.js:2:63758)
```

I'll sync to this PR and see if I can debug/run tests locally. There may be some other configurations for the locale data, and if so, we should think about simplifying and documenting!

**Comment by technohippy** [comment]
Sorry, some other files may have to be modified. I added some fixes but they must not enough.

**Comment by technohippy** [comment]
I'll try to fix these issues this weekends.

**Comment by technohippy** [comment]
@kepae I don't have cypress account but AIID in ja seems run successfully on my laptop anyway.

**Review by kepae — CHANGES_REQUESTED** [review]
Thanks for updating all of these various pieces, this PR will help make documentation on contributing language support.

For now, tests are still failing, and I can look into them repoducing them locally.
For running tests, a cypress cloud account isn't necessary -- any test can be run locally as well!

**Comment by technohippy** [comment]
Thank you for running tests for me. Yes, I can run tests on my laptop but they throw a lot of 401 errors so I gave up to run tests. I'll investigate more. Anyway I fixed a url and applied your change.

**Review by kepae — COMMENTED** [review]
Circling back here with new 2024 energy -- hope you've had a good year so far @technohippy !

We have since implemented a github workflow to build + test changes and in parallel deploy them (not blocking on tests), and are working on making this easier for contributors.

Until that's perfected -- @pdcp1 would you please take a look at these changes and see if you can build+test this locally? and provide any guidance on the translation changes?

**Comment by pdcp1** [comment]
Sure, I will take a look

**Comment by pdcp1** [comment]
@technohippy Could you please merge `staging` branch into this one to pull the last changes? We made some improvements on the tests and deployment.
In the meantime, I am performing tests to determine the reason for the failed tests and will update you soon.

**Review by pdcp1 — CHANGES_REQUESTED** [review]

**Comment by technohippy** [comment]
@pdcp1 Thansk, I'll fix them all this weekends.

File: site/gatsby-site/cypress/e2e/integration/citeEdit.cy.js [hunk 1]
```diff
@@ -743,6 +743,20 @@ describe('Edit report', () => {
         );
       });
 
+    cy.wait('@updateOneReportTranslation')
+      .its('request.body.variables')
+      .then((variables) => {
+        expect(variables.input.title).to.eq('Este es el Titulo en japonés');
+        expect(variables.input.text).to.eq(
+          'Este es el texto que tiene un largo mayor a ochenta caracteres en frances!'
```
[the remark this answers is not in the document]
Comment (technohippy) [reply site/gatsby-site/cypress/e2e/integration/citeEdit.cy.js]: done

File: site/gatsby-site/cypress/e2e/integration/citeEdit.cy.js [hunk 2]
```diff
@@ -743,6 +743,20 @@ describe('Edit report', () => {
...
+        );
+        expect(variables.input.language).to.eq('ja');
+        expect(variables.input.report_number).to.eq(23);
+        expect(variables.input.plain_text).to.eq(
+          'Este es el texto que tiene un largo mayor a ochenta caracteres en frances!\n'
```
[the remark this answers is not in the document]
Comment (technohippy) [reply site/gatsby-site/cypress/e2e/integration/citeEdit.cy.js]: done

File: site/gatsby-site/cypress/e2e/unit/pageCreators/createReportPages.cy.js [hunk 3]
```diff
@@ -42,6 +42,14 @@ const languages = [
...
     dateFormat: 'DD-MM-YYYY',
   },
+  {
+    code: 'ja',
+    hrefLang: 'ja',
+    name: 'Japanese',
+    localName: '日本語',
+    langDir: 'ltr',
+    dateFormat: 'YYYY/MM/DD'
+  },
```
[the remark this answers is not in the document]
Comment (technohippy) [reply site/gatsby-site/cypress/e2e/unit/pageCreators/createReportPages.cy.js:52]: done

File: site/gatsby-site/cypress/e2e/unit/pageCreators/createCitationPages.cy.js [hunk 4]
```diff
@@ -79,6 +79,14 @@ const languages = [
...
     dateFormat: 'DD-MM-YYYY',
   },
+  {
+    code: 'ja',
+    hrefLang: 'ja',
+    name: 'Japanese',
+    localName: '日本語',
+    langDir: 'ltr',
+    dateFormat: 'YYYY/MM/DD'
+  },
```
[the remark this answers is not in the document]
Comment (technohippy) [reply site/gatsby-site/cypress/e2e/unit/pageCreators/createCitationPages.cy.js:89]: done

**Review by pdcp1 — CHANGES_REQUESTED** [review]
A new translation file `/gatsby-site/i18n/locales/en/incidents.json` was added to the project. Please copy that file into the `/gatsby-site/i18n/locales/ja` folder.
If your fork is not up to date you should merge Staging again.

**Pull request closed without merging**